llvm/lib/Target/AMDGPU/AMDGPUInsertSingleUseVDST.cpp

```diff
@@ -114,6 +102,22 @@ class AMDGPUInsertSingleUseVDST : public MachineFunctionPass {
           for (const MCRegUnit &Unit : RegUnits)
             RegisterUseCount.erase(Unit);
         }
+
+        for (const auto &Operand : MI.all_uses()) {
```

Comment:
What happens with regmask operands? What happens with call uses?

Reply:
From the point of view of this pass, I'm not sure these need special handling. The pass will treat them as regular instructions, just with a large amount of registers used.

Reply:
I guess you should disable all optimization across a call, like you do across exec mask changes, but that can be a separate patch.